using_restricted_array.rb

```diff
@@ -8,28 +8,88 @@
 
 ## Calculates the length of the restricted integer array_size
 def length(array)
-  puts "NOT IMPLEMENTED"
+    # have a counter that keeps track of where in the array you are
+    i = 0;
+    # loop through the array and check if that number is equal to nil
+    while (array[i] != nil)
+    # if it is not nil, increase i
+        i++
+    end
+
+    # if it is nil, then the length is i
+    return i
+
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+    i = 0;
+    while (array[i] != 0)

```

Comment:
Compare to nil and not 0, like you did with length method. 0 could be a legitimate value.